javagen/src/main/java/com/azure/autorest/Javagen.java

```diff
@@ -75,206 +75,217 @@ public boolean processInternal() {
+            protected NodeTuple representJavaBeanProperty(Object javaBean, Property property, Object propertyValue,
+                Tag customTag) {
+                // if value of property is null, ignore it.
+                if (propertyValue == null) {
+                    return null;
                 }
-
-                for (AsyncSyncClient syncClient : syncClients) {
-                    javaPackage.addSyncServiceClient(syncClient.getPackageName(), syncClient);
+                else {
+                    return super.representJavaBeanProperty(javaBean, property, propertyValue, customTag);
                 }
             }
+        };
 
-            // Method group
-            for (MethodGroupClient methodGroupClient : client.getServiceClient().getMethodGroupClients()) {
-                javaPackage.addMethodGroup(methodGroupClient.getPackage(), methodGroupClient.getClassName(), methodGroupClient);
-                if (settings.shouldGenerateClientInterfaces()) {
-                    javaPackage.addMethodGroupInterface(methodGroupClient.getInterfaceName(), methodGroupClient);
-                }
-            }
+        LoaderOptions loaderOptions = new LoaderOptions();
+        loaderOptions.setMaxAliasesForCollections(Integer.MAX_VALUE);
+        Yaml newYaml = new Yaml(new Constructor(loaderOptions), representer, new DumperOptions(), loaderOptions);
+        CodeModel codeModel = newYaml.loadAs(file, CodeModel.class);
+        return codeModel;
+    }
 
-            // Sample
-            if (settings.isLowLevelClient() && settings.isGenerateSamples()) {
-                Set<String> protocolExampleNameSet = new HashSet<>();
-
-                syncClients.stream().filter(c -> c.getMethodGroupClient() != null)
-                        .forEach(c -> c.getMethodGroupClient().getClientMethods().stream()
-                        .filter(m -> m.getType() == ClientMethodType.SimpleSyncRestResponse || m.getType() == ClientMethodType.PagingSync)
-                        .forEach(m -> {
-                            if (m.getProxyMethod().getExamples() != null) {
-                                m.getProxyMethod().getExamples().forEach((name, example) -> {
-                                    String filename = CodeNamer.toPascalCase(CodeNamer.removeInvalidCharacters(name));
-                                    if (!protocolExampleNameSet.contains(filename)) {
-                                        ProtocolExample protocolExample = new ProtocolExample(m, c, client.getServiceClient(), builderName, filename, example);
-                                        javaPackage.addProtocolExamples(protocolExample);
-                                        protocolExampleNameSet.add(filename);
-                                    }
-                                });
-                            }
-                        }));
-            }
+    private JavaPackage writeToTemplates(JavaSettings settings, CodeModel codeModel, Client client) {
```

Comment:
if considering unit test, making the function package-private instead of private may be better for unit test?

Reply:
Of course, will do that in unit testing.

Reply:
Changed, thanks for pointing out!